src/IncludedCommands/Persistent/pban.luau

```diff
@@ -1,46 +1,138 @@
+
+            local PersistentBans = Api.CommandData.PersistentBans :: PersistentBans.PersistentBans
+
+            local Success, Response, BannedIds, ImmuneIds = PersistentBans:BanIds({
+                ["UserIds"] = Ids,
+                ["Duration"] = Duration,
+                ["DisplayReason"] = DisplayReason,
+                ["PrivateReason"] = PrivateReason,
+                ["ApplyToUniverse"] = ExperienceWide,
+                ["ExcludeAltAccounts"] = not IncludeAlts,
+            }, Player.UserId, Player.Name)
+
+            SendResponseRemote:FireClient(Player, "Ban", Success, Response, BannedIds, ImmuneIds)
+
+            VarStore['DisplayReason'] = nil
+            VarStore['PrivateReason'] = nil
+        end)
+    end,
+    ClientLoad = function(Api: Types.NexusAdminApiClient)
+        local Executor = Api.Executor :: Types.Executor
+        local Messages = Api.Messages :: Types.MessagesClient
+
+        local DisplayReasonMessage = 'run edit ${var .DisplayReason} \\\\\n && var= .DisplayReason ||'
+        local PrivateReasonMessage = 'run edit ${var .PrivateReason} \\\\\n && var= .PrivateReason ||'
+
+        local GetReasonsRemote = IncludedCommandUtil:GetRemote("PersistentBan") :: RemoteEvent
+
+        GetReasonsRemote.OnClientEvent:Connect(function(...)
+            Messages:DisplayHint("Persistent Bans: Please provide an optional Display Ban Message. This will be shown to the banned individual(s). 400 Character Limit.",10)
+            Executor:ExecuteCommandWithOrWithoutPrefix(DisplayReasonMessage, Players.LocalPlayer)
+
+            Messages:DisplayHint("Persistent Bans: Please provide an optional Private Ban Message. This will be visible only to user(s) authorized to access :pbans. 950 Character Limit.",10)
+            Executor:ExecuteCommandWithOrWithoutPrefix(PrivateReasonMessage, Players.LocalPlayer)
+
+            GetReasonsRemote:FireServer(...)
+        end)
     end,
-    ServerRun = function(CommandContext: Types.CmdrCommandContext, PlayerNames: {string}, Message: string?): string?
+    ServerRun = function(CommandContext: Types.CmdrCommandContext, PlayerNames: {string}, Duration: number, ExperienceWide: boolean, IncludeAlts: boolean): string?
+        local PersistentBans = require(script.Parent.Parent:WaitForChild("Resources"):WaitForChild("PersistentBans"))
+
         local Util = IncludedCommandUtil.ForContext(CommandContext)
         local Api = Util:GetServerApi()
 
         --Return if the persistent bans weren't initialized.
         local PersistentBans = Api.CommandData.PersistentBans :: PersistentBans.PersistentBans
         if not PersistentBans:WasInitialized() then
-            return "Persistent bans failed to initialize."
+            if Api.CommandData.PersistentBans.Enabled then
+                return {"Persistent bans failed to initialize."}
+            else
+                return {"RobloxBanAPI is disabled within this experience."}
```

Comment:
This has a typing issue due to `ServerRun` expecting a `string?` response, but this is a `{string}`. `string?` is correct.